feat:create-merge-accounts-selector

k-g-j opened the pull request:
## **Description**

Moves the exported `mergeAccounts` function out of the account-list-menu and into a selector where it can be imported and used in other files in a more standardized way. Also, refactors the current implementation of `mergeAccounts` for more efficient time complexity.

## **Related issues**

Fixes: This selector will be unnecessary and can be removed in the account-list-menu and snap-account-card after using one call to get accounts that will have all data without needing to merge two different account objects when #21709 is merged 

## **Manual testing steps**

N/A

### Jest

`jest selectors.test.js -t '#mergeAccounts'`

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

k-g-j committed:
Added new mergeAccounts selector and replaced previous call to mergeAccounts to use new selector implementation

k-g-j committed:
Fixed import code fencing in account-list-menu

k-g-j committed:
Removed TODOs

k-g-j committed:
Merge remote-tracking branch 'origin/develop' into feat/create-merge-accounts-selector

owencraston commented on ui/selectors/selectors.js:
```diff
@@ -381,6 +382,36 @@ export function getInternalAccountsSortedByKeyring(state) {
+export function mergeAccounts(state, accountsWithBalances) {
+  const internalAccounts = getInternalAccounts(state);
+  const internalAccountsMap = new Map(
+    internalAccounts.map((acc) => [acc.address, acc]),
```
Love this performance improvement

owencraston commented on ui/selectors/selectors.js:
```diff
@@ -381,6 +382,36 @@ export function getInternalAccountsSortedByKeyring(state) {
+ * @param accountsWithBalances - list of accounts with balances
+ * @returns merged accounts list with balances and internal account data
+ */
+export function mergeAccounts(state, accountsWithBalances) {
```
I think we can have a more descriptive name here. Perhaps something like getAllAccounts?

bowensanders committed:
[Snaps E2E] Create test for snaps home page (#22436)

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->
Creates a test for snaps home pages

## **Related issues**

Fixes: issue #22424 

## **Manual testing steps**

1. this can be run manually or as part of CI

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

mikesposito committed:
Add test for vault decryptor (#22317)

sahar-fehri committed:
feat: bump assets controllers (#22398)

## **Description**

Bumps assets-controllers version


<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>

sahar-fehri committed:
feat: fetch token fiat amount on import confirmation modal (#22263)

## **Description**

The goal of this PR is to fetch user token fiat balance on the import
modal confirmation page.
There is an ongoing PR on core to enable this:
https://github.com/MetaMask/core/pull/3657



## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMASSETS-88
also design wise relates to this PR
https://github.com/MetaMask/metamask-extension/pull/21704

## **Manual testing steps**

1. Go to home page
2. Click on import tokens button
3. Select a token
4. Click import
5. Notice the token balance in fiat on the confirmation page

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**


![image](https://github.com/MetaMask/metamask-extension/assets/10994169/a7513ded-2960-4c6d-972d-8bb1d673f0df)
### **After**


![image](https://github.com/MetaMask/metamask-extension/assets/10994169/1232ba09-9769-476d-b4f8-12415ac55f2c)


## **Pre-merge author checklist**

- [] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

k-g-j committed:
Merge branch 'develop' into feat/create-merge-accounts-selector

georgewrmarshall committed:
Renaming index.js to index.ts and exporting types from component-library (#22432)

## **Description**

This PR addresses renaming the `index.js` file in the component-library
to `index.ts` as part of our ongoing migration to TypeScript. All
components in the component-library have been successfully converted to
TypeScript, and renaming the `index.js` file to `index.ts` is the next
step in this process.

This change allows us to leverage TypeScript's static typing in the
index file and ensures consistency in our codebase. Additionally, this
PR enables the export of component types directly from the index file,
making it easier for engineers to import types.

## **Related issues**

Fixes: #22430

## **Manual testing steps**

1. Pull this branch 
2. Navigate to the component-library directory.
3. Check that the index.js file has been renamed to index.ts.
4. Go to a TypeScript file. The below recording shows
`ui/components/multichain/import-token-link/import-token-link.tsx`
5. Try importing types from the component-library e.g.
`BoxProps`,`ButtonLinkProps`, etc

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**


https://github.com/MetaMask/metamask-extension/assets/8112138/f1327423-f154-4457-b930-1341f4e19dcf

### **After**



https://github.com/MetaMask/metamask-extension/assets/8112138/acb6a438-4e74-41a6-8253-fc81eb6eefa4



## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

georgewrmarshall committed:
Fixing dark mode in storybook docs view (#21798)

## **Description**
This PR updates the Storybook configuration to enable Dark Mode in the
Docs view, which was previously non-functional. It also removes
unnecessary files from the codebase for better code maintainability.

## **Related issues**

Fixes: #19671 

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Turn dark mode on and off buy pressing the 🌞 / 🌙 icon in the top
toolbar
3. See that dark mode works for docs pages

## **Screenshots/Recordings**

### **Before**



https://github.com/MetaMask/metamask-extension/assets/8112138/10de57f4-c527-423b-a1b6-8b96f568b895


### **After**



https://github.com/MetaMask/metamask-extension/assets/8112138/5bded120-70ea-44fa-8791-6b7f2ec352c9



## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>

georgewrmarshall committed:
Fixing deprecation messages for BUTTON_VARIANT and BUTTON_SIZES (#22451)

## **Description**
This PR addresses an issue with the deprecation messages for the
`BUTTON_VARIANT` and `BUTTON_SIZES` constants. Currently, the messages
are misplaced and do not correspond to the correct constants. This
update ensures that each deprecation message is correctly placed above
its respective deprecated constant, improving clarity and accuracy.

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Pull this branch
2. Search for `BUTTON_VARIANT`
3. Hover over the constant to see the correct deprecation notice
4. Repeat for `BUTTON_SIZES`

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

![Screenshot 2024-01-08 at 8 52
55 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/98102aec-6b4b-4de7-96e0-db68e033ec50)
![Screenshot 2024-01-08 at 8 53
40 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/c8d3cfeb-e771-46de-a7b2-1eae9d810c61)


### **After**

![Screenshot 2024-01-08 at 8 54
16 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/1698971f-d7e0-47e1-a1ba-4c006da96d24)
![Screenshot 2024-01-08 at 8 54
41 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/8340fa4d-05e6-4363-b21c-22d4141a3cd2)


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

k-g-j committed:
Merge remote-tracking branch 'origin' into feat/create-merge-accounts-selector

k-g-j commented:
No longer relevant with accounts controller identities merging
